packages/forms/src/index.tsx

```diff
@@ -113,6 +116,59 @@ const inputTagProps = <T extends InputTagProps>(
   return tagProps
 }
 
+// A hook-like function merge field props and return props (ref, onChange and onBlur primarily) of RHF v7 register()
+const useFieldRegister = <
+  T extends Omit<ValidatableFieldProps, 'defaultValue'>,
+  E
+>(
+  props: T,
+  ref?: React.ForwardedRef<E>
+) => {
+  const { register } = useFormContext()
+  const validation = props.validation || { required: false }
+
+  // Primarily for TextAreaField
+  if (!validation.validate && props.transformValue === 'Json') {
+    validation.validate = jsonValidation
+  }
+
+  const tagProps = inputTagProps(props) as T & {
+    onChange?: React.ChangeEventHandler<T>
+    onBlur?: React.FocusEventHandler<T>
+  }
+  const {
+    ref: _ref,
+    onBlur: handleBlur,
+    onChange: handleChange,
+    ...rest
+  } = register(props.name, validation)
+
+  const onBlur: React.FocusEventHandler<T> = (event) => {
+    handleBlur(event)
+    tagProps?.onBlur?.(event)
+  }
+  const onChange: React.ChangeEventHandler<T> = (event) => {
+    handleChange(event)
+    tagProps?.onChange?.(event)
+  }
```

Comment:
Agree with you here that react hook form's blur/change handlers should be invoked first. Also falls in line with their [documentation](https://react-hook-form.com/api/useform/register) (might have to scroll down a bit to see the screenshot below):

![image](https://user-images.githubusercontent.com/32992335/127256746-376a43db-3b4b-414f-ab14-9edcffecdc83.png)